apis/python/src/tiledbsoma/soma_dataframe.py

```diff
@@ -169,9 +169,9 @@ def read(
         # TODO: batch_size
         # TODO: partition,
         # TODO: platform_config,
-    ) -> Iterator[pa.RecordBatch]:
+    ) -> Iterator[pa.Table]:
```

Comment:
why not type the `ids` (aka coordinates) parameter to something more specific?  See for example SOMADenseNdArray.

I think you can literally use the `SOMADenseCoordinates` type (in types.py) as this is a dense dataframe, which only takes a scalar or slice.

Reply:
Also, we need to decide on whether or not `ids` is optional.  The behavior is inconsistent between DataFrame and NdArray - in the former, None is "all", in the latter, None is not accepted (only the explicit `slice(None)` is accepted).

CC: @thetorpedodog - if you want to propose a standard convention, we can pull it through the entire package.

Reply:
If we want to make it not an Optional value, we can easily preserve the "get everything if unspecified" behavior by doing

`ids: Union[Sequence[int], slice] = slice(None)`

I like this solution because it ends up with an explicit default with already-specified semantics.

Reply:
@bkmartinjr @thetorpedodog thanks!

Optionality of `ids` is orthogonal to this PR which is solely `pa.RecordBatch` -> `pa.Table`, and has been open long enough, and rebased enough times -- I created https://github.com/single-cell-data/TileDB-SOMA/issues/374 to track that